pyQuARC/code/datetime_validator.py

```diff
@@ -143,13 +143,39 @@ def validate_datetime_against_granules(
 
         validity = True
         last_granule_datetime = None
-
+        date_time = None
+
+        # Define the formats in decreasing order of precision
+        formats = [
+            "%Y-%m-%dT%H:%M:%S.%f",  # Year to microsecond
+            "%Y-%m-%dT%H:%M:%S",  # Year to second
+            "%Y-%m-%dT%H:%M",  # Year to minute
+            "%Y-%m-%dT%H",  # Year to hour
+            "%Y-%m-%d",  # Year to day
+            "%Y-%m",  # Year to month
+            "%Y",  # Year
+        ]
+
+        # Function to determine the precision of a datetime string
+        def get_precision(dt_str):
+            for fmt in formats:
+                try:
+                    date_time = datetime.strptime(dt_str, fmt)
+                    return date_time, fmt
+                except ValueError:
+                    continue
+            return None, False
+
+        # Compare the precision of the two datetime strings
+        # return get_precision(datetime_str1) == get_precision(datetime_str2)
         if len(granules["feed"]["entry"]) > 0:
             last_granule = granules["feed"]["entry"][0]
             last_granule_datetime = last_granule.get(time_key)
-            validity = datetime == last_granule_datetime
+            date_time, date_time_format = get_precision(datetime_string)
```

Comment:
What is the use of `date_time_format`?

Reply:
I return that date_time_format to see in which format date_time is converting, but in this code, it doesn't have any use so, I will remove it. Thank you for addressing that.

Reply:
it is not updated?

Reply:
Updated.